### PR TITLE
Alasdair/ecom 2765 publish button

### PR DESCRIPTION
**Pull request by AlasdairSwan**
@rlucioni @jimabramson please review this, the first subtask for ECOM-2211 (subsequent PRs will add the confirmation modal and lock down the editable fields for not unpublished courses)

### CODE REVIEW

File: programs/static/templates/program_details.underscore [hunk 1]
```diff
@@ -1,48 +1,58 @@
...
-            <input type="text" value="<%- marketing_slug %>" id="program-marketing-slug" class="program-marketing-slug field-input is-hidden" name="marketing_slug" data-field="marketing_slug" maxlength="255">
-            <button class="js-enable-edit btn btn-edit">
-                <span class="icon-fallback icon-fallback-text">
-                    <span class="icon icon-pencil" aria-hidden="true"></span>
-                    <span class="text"><%= gettext('Edit the program\'s marketing slug.') %></span>
-                </span>
+        <div class="layout-col layout-col-a">
+          <% if ( status === 'unpublished' ) { %>
+            <button class="js-publish-program btn btn-default btn-base btn-grey right">
+                <span>Publish</span>
```
Comment (rlucioni) [inline programs/static/templates/program_details.underscore]: Mark for translation.

File: programs/static/js/views/program_details_view.js [hunk 2]
```diff
@@ -92,6 +107,11 @@ define([
                    .addClass( 'edit' )
                    .focus();
                 $btn.addClass( 'is-hidden' );
+            },
+
+            publishProgram: function() {
+                this.model.set( 'status', 'published' );
```
Comment (rlucioni) [inline programs/static/js/views/program_details_view.js]: The value configured on the server to indicate that a course has been published is "active."

File: programs/static/js/test/specs/program_details_spec.js [hunk 3]
```diff
@@ -263,6 +264,24 @@ define([
...
+                        defaultStatus = programData.status,
+                        publishedStatus = 'active';
+
+                    expect( view.model.get( 'status' ) ).toEqual( defaultStatus );
+                    expect( view.model.get( 'status' ) ).not.toEqual( publishedStatus );
+
+                    $btn.click();
+
+                    expect( view.model.set ).toHaveBeenCalled();
+                    expect( view.model.get( 'status' ) ).not.toEqual( defaultStatus );
```
Comment (jimabramson) [inline programs/static/js/test/specs/program_details_spec.js]: is this line unnecessary given what the next line does?

Comment (AlasdairSwan) [inline programs/static/js/test/specs/program_details_spec.js]: I suppose it does, I just wanted to make it explicit that ```defaultState !== publishedState```

**Comment by jimabramson** [comment]
:+1:

**Pull request merged**